pkg/globalmanager/controllers/jointinference/jointinferenceservice.go

```diff
@@ -422,113 +351,223 @@ func (c *Controller) createWorkers(service *sednav1.JointInferenceService) (acti
+		return
 	}
-	workerParam.Mounts = append(workerParam.Mounts, runtime.WorkerMount{
-		URL: &runtime.MountURL{
-			URL:                   cloudModel.Spec.URL,
-			Secret:                modelSecret,
-			DownloadByInitializer: true,
-		},
-		Name:    "model",
-		EnvName: "MODEL_URL",
-	})
 
-	workerParam.Env = map[string]string{
-		"NAMESPACE":    service.Namespace,
-		"SERVICE_NAME": service.Name,
-		"WORKER_NAME":  "cloudworker-" + utilrand.String(5),
+	c.addDeployment(curD)
+}
+
+// deleteDeployment enqueues the jointinferenceservice obj When a deleteDeployment is deleted
+func (c *Controller) deleteDeployment(obj interface{}) {
+	deployment, ok := obj.(*appsv1.Deployment)
 
-		"BIG_MODEL_BIND_PORT": strconv.Itoa(int(bigModelPort)),
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Warningf("couldn't get object from tombstone %+v", obj)
+			return
+		}
+		deployment, ok = tombstone.Obj.(*appsv1.Deployment)
+		if !ok {
+			klog.Warningf("tombstone contained object that is not a Deployment %+v", obj)
+			return
+		}
 	}
 
-	workerParam.WorkerType = jointInferenceForCloud
+	// If the deployment is accidentally deleted, recreate the deployment.
+	newDeployment := deployment.DeepCopy()
+	serviceName := func(input string) string {
+		return strings.Split(input, "-deployment")[0]
+	}(newDeployment.Name)
+	_, err := c.serviceLister.JointInferenceServices(newDeployment.Namespace).Get(serviceName)
+	if !errors.IsNotFound(err) {
+		// Remove unnecessary metadata.
+		newDeployment.ResourceVersion = ""
+		newDeployment.UID = ""
+		// Create a new deployment.
+		_, err := c.kubeClient.AppsV1().Deployments(newDeployment.Namespace).Create(context.TODO(), newDeployment, metav1.CreateOptions{})
+		if err != nil {
+			klog.Errorf("failed to recreate deployment %s: %v", deployment.Name, err)
+			return
+		}
+	}
 
-	// create cloud pod
-	_, err = runtime.CreatePodWithTemplate(c.kubeClient,
-		service,
-		&service.Spec.CloudWorker.Template,
-		&workerParam)
-	return err
+	klog.Infof("Successfully recreated deployment %s", deployment.Name)
+	c.enqueueByDeployment(newDeployment, true)
 }
 
-func (c *Controller) createEdgeWorker(service *sednav1.JointInferenceService, bigModelHost string, bigModelPort int32) error {
-	// deliver pod for edgeworker
-	ctx := context.Background()
-	edgeModelName := service.Spec.EdgeWorker.Model.Name
-	edgeModel, err := c.client.Models(service.Namespace).Get(ctx, edgeModelName, metav1.GetOptions{})
+func (c *Controller) updateInferenceServices(old, cur interface{}) error {
+	oldService := old.(*sednav1.JointInferenceService)
+	newService := cur.(*sednav1.JointInferenceService)
+	// Check the changes in specific fields and perform corresponding operations.
+	if !reflect.DeepEqual(oldService.Spec.CloudWorker, newService.Spec.CloudWorker) {
+		// If the cloud inference service changes, perform the corresponding update operation.
+		return c.updateCloudWorker(newService)
```

Comment:
Can't return directly; otherwise, the subsequent code won't be executed directly.


Modify the updateInferenceServices function as follows.
```go
func (c *Controller) updateInferenceServices(old, cur interface{}) error {
	oldService := old.(*sednav1.JointInferenceService)
	newService := cur.(*sednav1.JointInferenceService)

	// Check if the cloud worker configuration has changed
	if !reflect.DeepEqual(oldService.Spec.CloudWorker, newService.Spec.CloudWorker) {
		// Update cloud worker and log any errors
		if err := c.updateCloudWorker(newService); err != nil {
			klog.Errorf("Failed to update cloud worker for service %s/%s: %v", newService.Namespace, newService.Name, err)
		}
	}

	// Retrieve the address of the cloud inference service
	var bigModelHost string
	svc, err := c.kubeClient.CoreV1().Services(oldService.Namespace).Get(context.Background(),
		strings.ToLower(oldService.Name+"-"+jointInferenceForCloud), metav1.GetOptions{})
	if err != nil {
		if errors.IsNotFound(err) {
			// If the service does not exist, create a new one and retrieve its address
			klog.Info("Cloud service not found, creating new service...")
			bigModelHost, err = runtime.CreateEdgeMeshService(c.kubeClient, oldService, jointInferenceForCloud, BigModelPort)
			if err != nil {
				klog.Errorf("Failed to create EdgeMesh service for service %s/%s: %v", oldService.Namespace, oldService.Name, err)
			}
		} else {
			klog.Errorf("Failed to get cloud service %s/%s: %v", oldService.Namespace, oldService.Name, err)
		}
	} else {
		bigModelHost = fmt.Sprintf("%s.%s", svc.Name, svc.Namespace)
	}

	// Check if the edge worker configuration has changed
	if !reflect.DeepEqual(oldService.Spec.EdgeWorker, newService.Spec.EdgeWorker) {
		// Update edge worker and log any errors
		if err := c.updateEdgeWorker(newService, bigModelHost); err != nil {
			klog.Errorf("Failed to update edge worker for service %s/%s: %v", newService.Namespace, newService.Name, err)
		}
	}

	return nil
}
```

Reply:
modified.